packages/features/apps/AdminAppsList.tsx

```diff
@@ -232,22 +235,43 @@ const EditKeysModal: FC<{
                 key={key}
                 control={formMethods.control}
                 defaultValue={keys && keys[key] ? keys?.[key] : ""}
-                render={({ field: { value } }) => (
-                  <TextField
-                    label={key}
-                    key={key}
-                    name={key}
-                    value={value}
-                    onChange={(e) => {
-                      formMethods.setValue(key, e?.target.value);
-                    }}
-                  />
-                )}
+                render={({ field: { value } }) => {
+                  const isPasswordField =
+                    key.toLowerCase().includes("secret") ||
+                    key.toLowerCase().includes("password") ||
+                    key.toLowerCase().includes("key");
+                  const FieldComponent = isPasswordField ? PasswordField : TextField;
+
+                  return (
+                    <FieldComponent
+                      label={key}
+                      key={key}
+                      name={key}
+                      value={value}
+                      containerClassName="pb-5"
+                      hint={
+                        (appKeySchema.shape as Record<string, { description?: string } | undefined>)[key]
+                          ?.description
+                      }
+                      onChange={(e) => {
+                        formMethods.setValue(key, e?.target.value);
+                      }}
+                    />
+                  );
+                }}
               />
             ))}
           </Form>
         )}
-        <DialogFooter showDivider className="mt-8">
+        {!!docsUrl && (
+          <div className="text-subtle mb-3 w-full text-sm">
+            <a target="_blank" rel="noreferrer" className="hover:underline" href={docsUrl}>
+              <Icon name="book-open" className="-mt-1 mr-1 inline h-3 w-3" />
+              {t("documentation")}
+            </a>
+          </div>
+        )}
+        <DialogFooter noSticky showDivider className="mt-4">
```

Comment:
Hey, this dialog is not opened randomly. An admin will open this dialog only for setting the app credentials. We don't need to attach any documentation within the dialog; that's why we provide a README file to retrieve the credentials and set them accordingly

Reply:
Hey,
Thank you for reviewing. I understand. However, the README file is not even available in the UI. So, I believe it's a bit cumbersome to access for a non-developer. I'd be open to another solution though (e.g. making the README file available in the UI). Adding a link to a docs URL was just the easiest option to implement for me. What do you think?